lab/storefront-dropins/Lab.md

```diff
@@ -0,0 +1,478 @@
+          $include: ./actions/payment-method/actions.config.yaml
+```
+
+4. **Deploy your changes** to App Builder by running:
+
+```bash
+aio app deploy --force-build --force-deploy
+```
+
+5. After deployment, run:
+
+```bash
+aio app get-url
+```
+
+6. Confirm that both the following endpoints are listed:
+   - `payment-method/create-session`
+   - `payment-method/validate-payment`
+7. **Note:** You will need the URL for `validate-payment` in the next step.
+
+### Step 2.2: Subscribe to the Webhook
+1. Copy your validate-payment endpoint URL from Step 2.1 (the URL that ends with `/validate-payment`).
+
+2. Run these commands in your terminal (copy and paste each block):
+
+```bash
+# TODO Paste your validate-payment URL between the quotes
+VALIDATE_PAYMENT_URL=""
+```
+
+```bash
+WEBHOOK_JSON='
+{
+  "webhook": {
+    "webhook_method": "observer.sales_order_place_before",
+    "webhook_type": "before",
+    "batch_name": "validate_payment",
+    "hook_name": "oope_payment_methods_sales_order_place_before",
+    "url": "'$VALIDATE_PAYMENT_URL'",
+    "required": true,
+    "method": "POST",
+    "fields": [
+      {"name": "payment_method", "source": "data.order.payment.method"},
+      {"name": "payment_additional_information", "source": "data.order.payment.additional_information"}
+    ],
+    "rules": [
+      {"field": "data.order.payment.method", "operator": "equal", "value": "PARTNER-PAY"}
+    ]
+  }
+}'
+
+curl -s -X POST $REST_API/V1/webhooks/subscribe \
+  -H "Authorization: Bearer $ACCESS_TOKEN" \
+  -H "Content-Type: application/json" \
+  -d "$WEBHOOK_JSON" | jq .
+```
+
+### Step 2.3: Verify Webhook Subscription in Admin
+1. Log in to the Admin Panel:
+   `https://na1-sandbox.admin.commerce.adobe.com/<TENANT_ID>`
+2. Go to **System > Webhooks > Webhooks Subscriptions**.
+3. You should see your new webhook listed. Click "Select" to view its details.
+4. Confirm the following settings:
+
+   **Hook Settings**
+   | Setting | Value |
+   |---------|-------|
+   | Webhook Method | `observer.sales_order_place_before` |
+   | Webhook Type | `before` |
+   | Batch Name | `validate_payment` |
+   | Hook Name | `oope_payment_methods_sales_order_place_before` |
+   | URL | `https://<your-validate-payment-endpoint-url>` |
+   | Active | `Yes` |
+   | Method | `POST` |
+
+   **Hook Fields**
+   | Field Name | Source |
+   |------------|--------|
+   | payment_method | data.order.payment.method |
+   | payment_additional_information | data.order.payment.additional_information |
+
+   **Hook Rules**
+   | Field | Operator | Value |
+   |-------|----------|-------|
+   | data.order.payment.method | equal | PARTNER-PAY |
+
+### Step 2.4: Test Payment Validation
+1. Go to your storefront checkout page.
+2. Try to place an order using the "PARTNER-PAY" payment method.
+3. You should see an error message:
+   **"Invalid payment session"**
+4. This confirms that your webhook is active and the validation logic is working.
+
+> **Tips:**
```

Comment:
Maybe here you can add a tip about how to enable debugging for webhooks,
see: https://magento.slack.com/archives/C07UW14U0UD/p1748276393158309?thread_ts=1748258405.378849&cid=C07UW14U0UD
and
https://developer.adobe.com/commerce/extensibility/webhooks/responses/#database-logging